### ISSUE TITLE
All Short Calls_Network_V1 Error

### ISSUE THREAD

**Issue by hdesalvo** [DrCoffey/DeepSqueak#100]
This bug has been mentioned before in this forum, but I keep having issues with only one network. I checked the add-ons and they all are installed. 

The "All Short Calls_Network_V1.mat" network does not work. I get the error message "Error in Network, Skiping Audio Chunk."

Using a .wav file (35 min). Went to to All Short Calls_Network_V1.mat and select audio file. Pressed "detect calls."

Accompanying code:

`Error in Network, Skiping Audio Chunk
Warning: Error using vision.internal.cnn.validation.checkDetectionInputImage (line 35)
Expected I to be nonempty.

Error in fasterRCNNObjectDetector/parseDetectInputs (line 731)
            [sz,params.DetectionInputWasBatchOfImages] =
            vision.internal.cnn.validation.checkDetectionInputImage(...

Error in fasterRCNNObjectDetector/detect (line 519)
            params = this.parseDetectInputs(I, varargin{:});

Error in SqueakDetect (line 120)
            [bboxes, scores, Class] = detect(network, im2uint8(im),
            'ExecutionEnvironment','auto','NumStrongestRegions',Inf);

Error in multinetdect_Callback (line 77)
        Calls = [Calls;
        SqueakDetect(AudioFile,NeuralNetwork,handles.audiofiles(CurrentAudioFile).name,Settings(:,k),j,length(audioselections),networkname,handles.optimization_slider.Value)];

Error in gui_mainfcn (line 95)
        feval(varargin{:});

Error in DeepSqueak (line 31)
    gui_mainfcn(gui_State, varargin{:});

Error in
matlab.graphics.internal.figfile.FigFile/read>@(hObject,eventdata)DeepSqueak('multinetdect_Callback',hObject,eventdata,guidata(hObject),1) 
  In SqueakDetect (line 168)
  In multinetdect_Callback (line 77)
  In gui_mainfcn (line 95)
  In DeepSqueak (line 31)
  In matlab.graphics.internal.figfile.FigFile/read>@(hObject,eventdata)DeepSqueak('multinetdect_Callback',hObject,eventdata,guidata(hObject),1) 
No Calls found in: G155_box1_8.17.2020`

Apologies for bad formatting in this, I am not very experienced with Matlab but have attempted everything to the best of my ability in previous issue reports.

I have tried MATLAB 2018b, 2019a, and 2020b and get the same error on all.

**Comment by DrCoffey** [comment]
Hey, first thing to check is if you have the most up to date version of DeepSqueak and all of the required toolboxes. This error has a number of sources and that will help narrow it down. 

-Kevin

**Comment by hdesalvo** [comment]
Hi,
Thank you so much for your response. I just checked and here is what I have:
Using MATLAB 2020b. How do I check if I have the most up to date version of DeepSqueak?
Toolboxes:
Statistics & Machine Learning
Signal Processing
Parallel Computing
Image Processing
Deep Learning
Curve Fitting
Computer Vision

**Comment by DrCoffey** [comment]
That all looks right. The newest version is DeepSqueak version 2.6.2

Can you detect the example files with default settings?

**Comment by DrCoffey** [comment]
I looked at your original issue again and am a little confused. Do all of the other networks work except for the All Short Calls_Network? I have never seen that before. I might need to see the audio file in question to help you out.

**Comment by hdesalvo** [comment]
The latest version of DeepSqueak is installed. 
The example rat recordings detects calls, but I get a "merge boxes" error (for all networks) and I can't see the calls. I detailed this in another issue but here is the error I got:

Error in SqueakDetect (line 175)
Calls = merge_boxes(AllBoxes, AllScores, AllClass, AllPowers, audio_info, 1, score_cuttoff, 0);

Error in multinetdect_Callback (line 77)
        Calls = [Calls;
        SqueakDetect(AudioFile,NeuralNetwork,handles.audiofiles(CurrentAudioFile).name,Settings(:,k),j,length(audioselections),networkname,handles.optimization_slider.Value)];

Error in gui_mainfcn (line 95)
        feval(varargin{:});

Error in DeepSqueak (line 30)
    gui_mainfcn(gui_State, varargin{:});

Error in
matlab.graphics.internal.figfile.FigFile/read>@(hObject,eventdata)DeepSqueak('multinetdect_Callback',hObject,eventdata,guidata(hObject),1)
 
Error while evaluating UIControl Callback

So, in summary, it seems like the only network I get the "error in network, skiping audio chunk" error is for V1 Short Rat Calls and I get the merging boxes error for every other network.

**Comment by hdesalvo** [comment]
On another computer, the issue persists with only the short rat call network. I also do not get that merging boxes issue. Not sure why this is occurring.

**Comment by DrCoffey** [comment]
Ok, so on a different machine you can detect and load calls using all networks except short rat call network? That is bizarre, I have not seen that before but I think it is looking like a computer specific issue. My next thought is to check on your graphics driver. DeepSqueak works best on an Nvidia card with updated drivers.

**Comment by hdesalvo** [comment]
I thought it was bizarre too! I've never heard of this happening. Ok, so we have a basic graphics card: Intel(R) Graphics 530. We do not have a Nvidia card but I'm sure we could get one. 

I just updated the drivers and tried again. I also changed the upper frequency range and since I am sampling at 96kHz, I can only detect up to 43kHz?

No error message on the short rat call network. However, the merging boxes issue still persists on this network and other networks.

**Comment by hdesalvo** [comment]
I want to add that this also occurs in the sample files. 
Unrecognized function or variable 'h'.

Error in merge_boxes (line 47)
if ~any(Accepted); close(h); return; end

Error in SqueakDetect (line 175)
Calls = merge_boxes(AllBoxes, AllScores, AllClass, AllPowers, audio_info, 1, score_cuttoff, 0);

Error in multinetdect_Callback (line 77)
        Calls = [Calls;
        SqueakDetect(AudioFile,NeuralNetwork,handles.audiofiles(CurrentAudioFile).name,Settings(:,k),j,length(audioselections),networkname,handles.optimization_slider.Value)];

Error in gui_mainfcn (line 95)
        feval(varargin{:});

Error in DeepSqueak (line 30)
    gui_mainfcn(gui_State, varargin{:});

Error in
matlab.graphics.internal.figfile.FigFile/read>@(hObject,eventdata)DeepSqueak('multinetdect_Callback',hObject,eventdata,guidata(hObject),1)
 
Error while evaluating UIControl Callback.

>>

**Comment by DrCoffey** [comment]
DeepSqueak uses machine vision code designed to run on nVidia GPUs. If anyone you know has access to one, I would see if you can test if this is your issue before buying one.

And yes, to successfully capture a given frequency you must sample at twice that frequency. For USV research it is suggested to sample at 250 kHz, so you don't miss any very high frequency calls. Rats seldom call above 90 kHz, so 192kHz sampling is also acceptable. 96kHz will mean you will miss many calls that are in the 50 kHz range.

https://en.wikipedia.org/wiki/Nyquist_frequency

**Comment by hdesalvo** [comment]
Thanks. I will check with the nVidia.

Do you think the sampling rate is the source of the error?

**Comment by DrCoffey** [comment]
If you send me the offending file I can try it on my machine and let you know.

**Comment by hdesalvo** [comment]
Sure. It's attached here as a google drive link.

We use Raven Pro to record but they have a bug that caps sampling rate at
96kHz (as detailed before). I just figured out that this is a big problem
based on what you sent me. I'm a new graduate student and I appreciate you
sending me information. I'm trying to figure out why this is occurring and
how I can fix it because our microphones are supposed to sample at 200kHz.
 375_box3_D30_postlever.wav
<https://drive.google.com/file/d/1At93_-WBNV680eH14ZWIxhHuIR9JrdJW/view?usp=drive_web>

--
*Heather DeSalvo*
Temple University | Honors Program | Class of 2021
B.S./M.S. Student | *Neuroscience: Systems, Behavior and Plasticity | *College
of Liberal Arts
*Minor: Chemistry *| College of Science & Technology
President, DMAX Club at Temple University
heather.desalvo@temple.edu

**Comment by DrCoffey** [comment]
Hey, no problem. Getting started with USVs can be a pain. I was able to use detect this file from 20-45 kHz with both the short rat call and long call networks. My guess is that your computer hardware is causing your issues. 

One thing to look out for when recording is the continuous noise band you are seeing at 33 kHz. That is likely some electrical noise in the room/chamber and removing it would help the recordings. If you can remove the noise physically, you can set a amplitude threshold in DeepSqueak to remove it because it is quieter than calls.

There are some interesting 22s in here, you must have some grumpy rats :)

![Capture](https://user-images.githubusercontent.com/39605011/102172844-7ff3fe00-3e4e-11eb-8500-6722cc8ce43d.PNG)

**Comment by hdesalvo** [comment]
Thank you! I just tried it again with your settings and it worked. I’m thinking my frequency range was a problem and same with chunk length. I was adjusting it for long rat calls. Will close the thread. Thanks again!
